### PR TITLE
Quality of Life on client.extraction_pipelines.runs.list

### PR DESCRIPTION
**Pull request by doctrino**
## Description
A small QoL improvement to the `client.extraction_pipelines.runs.list` method.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/extractionpipelines.py [hunk 1]
```diff
@@ -315,6 +317,79 @@ def create(
...
 
+    def filter(
+        self,
+        external_id: str,
+        status: Literal["success", "failure", "seen"] | Sequence[Literal["success", "failure", "seen"]] | None = None,
+        created_time: TimestampRange | Dict[str, Any] | str | None = None,
+        message: str | None = None,
+        limit: int = LIST_LIMIT_DEFAULT,
+    ) -> ExtractionPipelineRunList:
+        """`Filter extraction pipeline runs <https://developer.cognite.com/api#tag/Extraction-Pipelines-Runs/operation/filterRuns>`_
```
Comment (haakonvt) [inline cognite/client/_api/extractionpipelines.py:328]: It's hard to see exactly how this is different from `def list(` on line 227, which according to the docstring also targets `Extraction-Pipelines/operation/filterRuns` 🤔

**Comment by haakonvt** [comment]
I see `test_full` failed on windows... due to bad use of RNG in test data (written by me 😉 ), so I decided to fix that once and for all: https://github.com/cognitedata/cognite-sdk-python/pull/1323

File: cognite/client/_api/extractionpipelines.py [hunk 2]
```diff
@@ -247,12 +260,23 @@ def list(
                 >>> from cognite.client import CogniteClient
                 >>> c = CogniteClient()
                 >>> runsList = c.extraction_pipelines.runs.list(external_id="test ext id", statuses=["seen"], statuslimit=5)
+
+            Get all failed pipeline runs the last 24 hours for pipeliene 'extId':
+
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import ExtractionPipelineRun
+                >>> c = CogniteClient()
+                >>> res = c.extraction_pipelines.runs.filter(external_id="extId", statuses="failure", created_time="24h-ago")
```
Comment (doctrino) [inline cognite/client/_api/extractionpipelines.py:269]:
```suggestion
                >>> res = c.extraction_pipelines.runs.list(external_id="extId", statuses="failure", created_time="24h-ago")
```

File: tests/tests_integration/test_api/test_extraction_pipelines.py [hunk 3]
```diff
@@ -33,6 +36,28 @@ def new_extpipe(cognite_client):
     assert cognite_client.extraction_pipelines.retrieve(extpipe.id) is None
 
 
+@pytest.fixture
+def populated_runs(cognite_client: CogniteClient, new_extpipe: ExtractionPipeline) -> ExtractionPipelineRunList:
+    now = datetime_to_ms(datetime.now(timezone.utc))
+    a_year_ago = datetime_to_ms(datetime.now(timezone.utc).replace(year=datetime.now(timezone.utc).year - 1))
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_extraction_pipelines.py:42]: Didn't you create a nice helper for this already 😉 
```suggestion
    now = datetime_to_ms(dt_now := datetime.now(timezone.utc))
    a_year_ago = datetime_to_ms(YearAligner.add_units(dt_now, -1))
```

File: tests/tests_integration/test_api/test_extraction_pipelines.py [hunk 4]
```diff
@@ -33,6 +36,28 @@ def new_extpipe(cognite_client):
     assert cognite_client.extraction_pipelines.retrieve(extpipe.id) is None
 
 
+@pytest.fixture
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_extraction_pipelines.py:39]: Let's add some broader scoping here to avoid re-running for each test using it

Comment (doctrino) [reply tests/tests_integration/test_api/test_extraction_pipelines.py]: It is only used once, and due to the scope of `new_extpipe`, this cannot be made broader without rewriting all the tests below.

File: cognite/client/_api/extractionpipelines.py [hunk 5]
```diff
@@ -217,23 +225,28 @@ class ExtractionPipelineRunsAPI(APIClient):
...
+            created_time (dict[str, Any] | TimestampRange | str | None): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
             limit (int | None): Maximum number of ExtractionPipelines to return. Defaults to 25. Set to -1, float("inf") or None to return all items.
 
         Returns:
             ExtractionPipelineRunList: List of requested extraction pipeline runs
 
+        Tip:
+            The created_time paremeter supports in addition to a dictonary with the format
+            `{"min": epoch_min, "max": epoch_max}`, arguments given on the format `<integer>(s|m|h|d|w)-ago`.
+            For example, `12h-ago`, which will be parsed to `{"min"= now - 12h-ago}`.
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/_api/extractionpipelines.py:247]: @doctrino You resolved this one, but I still think the rewording is worth it. At least fix the typos.

File: cognite/client/_api/extractionpipelines.py [hunk 6]
```diff
@@ -247,12 +259,21 @@ def list(
                 >>> from cognite.client import CogniteClient
                 >>> c = CogniteClient()
                 >>> runsList = c.extraction_pipelines.runs.list(external_id="test ext id", statuses=["seen"], statuslimit=5)
+
+            Get all failed pipeline runs the last 24 hours for pipeliene 'extId':
```
Comment (haakonvt) [inline cognite/client/_api/extractionpipelines.py:263]:
```suggestion
            Get all failed pipeline runs in the last 24 hours for pipeline 'extId':
```

File: tests/tests_integration/test_api/test_extraction_pipelines.py [hunk 7]
```diff
@@ -33,6 +37,28 @@ def new_extpipe(cognite_client):
...
+            extpipe_external_id=new_extpipe.external_id,
+            status="success",
+            message="dolor sit amet",
+            created_time=a_year_ago,
+        ),
+    ]
+    created = []
+    for run in runs:
+        new_run = cognite_client.extraction_pipelines.runs.create(run)
+        created.append(new_run)
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_extraction_pipelines.py:58]: 🏌️-opportunity

**Review by haakonvt — APPROVED** [review]
Lgtm, 2 comment-related comments

**Pull request merged**